tests/FakeItEasy.Specs/OrderedCallMatchingSpecs.cs

```diff
@@ -204,33 +258,52 @@ public interface IHaveOneGenericParameter
-                .x(() => A.CallTo(() => fake.Bar(2)).MustHaveHappened().InOrder(sequentialCallContext));
+            "And then a call with argument 1 to have been made exactly once"
+                .x(() => exception = Record.Exception(() => lastAssertion.Then(A.CallTo(() => fake.Bar(1)).MustHaveHappened(Repeated.Exactly.Once))));
 
-            "And I expect the call with argument 1 to have been made next exactly once"
-                .x(() => exception = Record.Exception(() => A.CallTo(() => fake.Bar(1)).MustHaveHappened(Repeated.Exactly.Once).InOrder(sequentialCallContext)));
-
-            "Then the assertion should fail"
+            "Then the last assertion should fail"
                 .x(() => exception.Should().BeAnExceptionOfType<ExpectationException>());
 
-            "And the error should tell us that the call to Bar(1) was found too many times"
+            "And the error should say that the call to Bar(1) was found too many times"
                 .x(() => exception.Message.Should().Be(@"
 
   Assertion failed for the following call:
-    FakeItEasy.Specs.OrderedCallMatchingSpecs+IHaveNoGenericParameters.Bar(1)
+    FakeItEasy.Specs.OrderedCallMatchingSpecs+IFoo.Bar(1)
   Expected to find it exactly once but found it #2 times among the calls:
-    1: FakeItEasy.Specs.OrderedCallMatchingSpecs+IHaveNoGenericParameters.Bar(baz: 1)
-    2: FakeItEasy.Specs.OrderedCallMatchingSpecs+IHaveNoGenericParameters.Bar(baz: 2)
-    3: FakeItEasy.Specs.OrderedCallMatchingSpecs+IHaveNoGenericParameters.Bar(baz: 1)
+    1: FakeItEasy.Specs.OrderedCallMatchingSpecs+IFoo.Bar(baz: 1)
+    2: FakeItEasy.Specs.OrderedCallMatchingSpecs+IFoo.Bar(baz: 2)
+    3: FakeItEasy.Specs.OrderedCallMatchingSpecs+IFoo.Bar(baz: 1)
 
 "));
         }
 
-        public class Generic<T>
+        // Reported as issue 182 (https://github.com/FakeItEasy/FakeItEasy/issues/182).
+        [Scenario]
+        public static void OrderedAssertionOnInterfacesWithCommonParent(ISomethingBaz baz, ISomethingQux qux, IAfterMustHaveHappenedConfiguration lastAssertion, Exception exception)
         {
+            "Given a Fake baz"
+                .x(() => baz = A.Fake<ISomethingBaz>());
+
+            "And a Fake qux implementing an interface in common with baz"
+                .x(() => qux = A.Fake<ISomethingQux>());
+
+            "And a call on qux"
+                .x(() => qux.QuxMethod());
+
+            "And a call on baz"
+                .x(() => baz.BazMethod());
+
+            "When I expect the qux call to have been made"
+                .x(() => lastAssertion = A.CallTo(() => qux.QuxMethod()).MustHaveHappened());
+
+            "And I express the same expectation again"
+                .x(() => exception = Record.Exception(() => lastAssertion.Then(A.CallTo(() => qux.QuxMethod()).MustHaveHappened())));
+
+            "Then the expectation should fail"
+                .x(() => exception.Should().BeAnExceptionOfType<ExpectationException>());

```

Comment:
"not be met" is better with an expectation, but I think we should adopt the ubiquitous language and switch to "assertion" everywhere. Assuming we are all in agreement, then I think saying that "The assertion should fail" is acceptable, yes? (Likewise "pass" when things go better.)